kernel/vmemitcodes.ml

```diff
@@ -54,13 +54,74 @@ module RelocTable = Hashtbl.Make(struct
   let hash = hash_reloc_info
 end)
 
+module Positions :
+sig
+  type t
+  val of_list : int list -> t
+  val iter : (int -> unit) -> t -> unit
+end =
+struct
+
+type t = string
+(* Represent an ordered set of 32-bit integers as an array of successive diffs.
+   We use furthermore an approach where smaller integers use less bytes. Numbers
+   smaller than 255 are stored into one byte. This is a cheap way to compact
+   this data. *)
+
+let output buf n =
+  if n < 0xFF then Buffer.add_uint8 buf n
+  else
+    let () = Buffer.add_uint8 buf 0xFF in
```

Comment:
Since `n` is never zero (you assert it below), you could use zero instead of 255 as the sentinel value. That way, even `n = 255` would be stored on one byte.

Reply:
Indeed, will do.

Reply:
Actually the very first integer might still be 0, so this encoding won't work.

Reply:
No, position zero is necessarily the start of an instruction. There cannot be any relocation there. There are even parts of the code that rely on this property (e.g., `coq_tcode_of_code`).

You can either assert it, or if you really believe that one day we will have a relocation stored at position zero, you can use the escape byte to use the 32-bit encoding of zero.

Reply:
Done.